spark/src/main/scala/geotrellis/spark/io/file/geotiff/FileIMGeoTiffAttributeStore.scala

```diff
@@ -1,14 +1,26 @@
 package geotrellis.spark.io.file.geotiff
 
 import geotrellis.spark.io.hadoop.geotiff._
+
 import org.apache.hadoop.conf.Configuration
+import spray.json._
+import spray.json.DefaultJsonProtocol._
 
+import java.io.PrintWriter
 import java.net.URI
 
 object FileIMGeoTiffAttributeStore {
   def apply(
     name: String,
     uri: URI
-  ): InMemoryGeoTiffAttributeStore =
-    InMemoryGeoTiffAttributeStore(() => GeoTiffMetadataTree.fromGeoTiffMetadataList(HadoopGeoTiffInput.list(name, uri, new Configuration())))
+  ): InMemoryGeoTiffAttributeStore = {
+    def getDataFunction = () => HadoopGeoTiffInput.list(name, uri, new Configuration())
+    new InMemoryGeoTiffAttributeStore(() => GeoTiffMetadataTree.fromGeoTiffMetadataList(getDataFunction())) {
+      def persist(uri: URI): Unit = {
```

Comment:
`GeoTiffMetadataTree.fromGeoTiffMetadataList` asking for `List` is too restrictive, `Seq` or `TraversableOnce` is better, although I can't seem to consistently decide if `TraversableOnce` is preferable.